drivers/wifi/esp_at/esp.c

```diff
@@ -276,8 +277,19 @@ MODEM_CMD_DEFINE(on_cmd_cwlap)
 	memcpy(res.ssid, argv[1], i);
 	res.ssid_length = i;
 	res.rssi = strtol(argv[2], NULL, 10);
-	res.channel = strtol(argv[3], NULL, 10);
 
+#if defined(CONFIG_WIFI_ESP_AT_SCAN_MAC_ADDRESS)
```

Comment:
You could use this instead:
```
if (IS_ENABLED(CONFIG_WIFI_ESP_AT_SCAN_MAC_ADDRESS)) {

} else {

}
```
That way the compiler will compile the code regardless of the configuration. It's specifically important, as there doesn't seem to be any test added that enables `CONFIG_WIFI_ESP_AT_SCAN_MAC_ADDRESS`.

Reply:
Done.